src/string.cr

```diff
@@ -2405,25 +2404,47 @@ class String
   def compare(other : String, case_insensitive = false)
     return self <=> other unless case_insensitive
 
-    reader1 = Char::Reader.new(self)
-    reader2 = Char::Reader.new(other)
-    ch1 = reader1.current_char
-    ch2 = reader2.current_char
+    if ascii_only? && other.ascii_only?
+      size_cmp = self.bytesize <=> other.bytesize
+      return size_cmp if size_cmp != 0
```

Comment:
Quoting @asterite (from earlier comments) about this condition:
> Hey, I think this condition is wrong: if a string is longer/shorter than other it doesn't mean it's lower/greater. For example "z" is greater than "hello", while "h" is lower than "zzz".

Reply:
Oh I think this can be fixed by simply making it:
```cr
cmp = self <=> other
return cmp if cmp != 0
```
And that means all the specs are wrong too, right? Here's the first one:
```cr
 "fo".compare("FOO", case_insensitive: true).should eq(-1)
```
Here it says `-1` but actually `"fo" <=> "FOO"` returns `1` which this method should actually return.